lib/trip_dispatcher.rb

```diff
@@ -90,6 +93,58 @@ def load_trips
       trips
     end
 
+    # addition that takes passenger_id to create new trip
+    def request_trip(passenger_id)
+      check_id(passenger_id)
+      id = trips.length + 1
```

Comment:
For this exercise this works fine, but if the list of trips could have deleted entries, then setting the id this way could lead to duplicate ids.  

I would suggest instead:
```ruby
id = trips.max_by { |trip| trip.id }.id + 1
```